pkg/utils/download.go

```diff
@@ -48,33 +74,119 @@ func DownloadFromTarGzURL(URL string, destination string) error {
 	return err
 }
 
+func getURLToDownloadReleaseAsset(URL string, gitCredentials GitCredentials) (string, error) {
+	URLSlice := strings.Split(URL, "/")
```

Comment:
Would it be safer to parse this as a URL and use `URL.Hostname` see: https://golang.org/pkg/net/url/#URL.Hostname
We're already using the `net/url` package elsewhere in download.go.
This probably won't quite give you what you want if there's a port in the hostname for example.

Reply:
okay. this is just a private helper function, and we do parse this as a URL above, so I think we're quite safe

But I agree this function would be more reusable if we handled a URL type here, so I can make this change